cmd/gateway-handlers.go

```diff
@@ -186,6 +186,14 @@ func (api gatewayAPIHandlers) PutObjectHandler(w http.ResponseWriter, r *http.Re
 	bucket = vars["bucket"]
 	object = vars["object"]
 
+	// Validate storage class metadata if present
+	if _, ok := r.Header[amzStorageClassCanonical]; ok {
```

Comment:
looks like you can avoid declaring  amzStorageClassCanonical if you use:
r.Header.Get(amzStorageClass)

Reply:
The reason to do this way is to verify if the x-amz-storage-class is indeed set. Just using Headers.Get would be empty for both conditions i.e 

- when x-amz-storage-class is set but has empty value
- when x-amz-storage-class is not set

In a scenario like this we need to return error for first condition but not for second.